### ISSUE TITLE
bug in conda-forge's torchvision, version 0.10.1

### ISSUE THREAD

**Issue by ngam** [conda-forge/torchvision-feedstock#32]
- [x] I read [the conda-forge documentation](https://conda-forge.org/docs/user/introduction.html#how-can-i-install-packages-from-conda-forge) and could not find the solution for my problem there.

Issue: Your `torchvision` has a bug in the `torchvision.io.read_image` function. On MacOS ARM64, I get the following error: `RuntimeError: No such operator image::read_file`. On Linux x86, I still get a similar error most of the time, however sometimes with more clarity especially inside the official and latest pytorch container from NVIDIA (which relies on conda-forge): `RuntimeError: Wrong JPEG library version: library is 80, caller expects 62`. This hints at the jpeg decoder. 

Anyway, a clean reinstall from the pytorch channel solves this problem on Linux x86, but the pytorch channel has only 0.2.2 as the latest `torchvision` package for ARM64, which is not suitable. Therefore, conda-forge's `torchvision` still breaks... Could you look into this? Below is a walkthrough on ARM64, but similar walkthrough could be achieved on a Linux x86 inside the NVIDIA container or outside of it (as long as `torchvision` is from conda-forge). 

Or, could you fast-forward to the latest `torchvision` from pytorch? That would be 0.11.1. Thanks! 

```python
Python 3.9.7 | packaged by conda-forge | (default, Sep 29 2021, 19:24:02) 
[Clang 11.1.0 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.

>>> 
>>> import torch
>>> import torchvision
>>> 
>>> torchvision.io.read_image('... .jpg')
...
... line 63, in __getattr__
    op = torch._C._jit_get_operation(qualified_op_name)
RuntimeError: No such operator image::read_file

```

<br/>
Environment (<code>conda list</code>):
<details>

```bash
$ conda list

# Name                    Version                   Build  Channel
anyio                     3.3.4            py39h2804cbe_0    conda-forge
appnope                   0.1.2            py39h2804cbe_1    conda-forge
argon2-cffi               21.1.0           py39h5161555_0    conda-forge
async_generator           1.10                       py_0    conda-forge
attrs                     21.2.0             pyhd8ed1ab_0    conda-forge
babel                     2.9.1              pyh44b312d_0    conda-forge
backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
backports                 1.0                        py_2    conda-forge
backports.functools_lru_cache 1.6.4              pyhd8ed1ab_0    conda-forge
bleach                    4.1.0              pyhd8ed1ab_0    conda-forge
brotlipy                  0.7.0           py39h5161555_1001    conda-forge
ca-certificates           2021.10.8            h4653dfc_0    conda-forge
certifi                   2021.10.8        py39h2804cbe_0    conda-forge
cffi                      1.14.6           py39h52b1de0_1    conda-forge
chardet                   4.0.0            py39h2804cbe_1    conda-forge
charset-normalizer        2.0.0              pyhd8ed1ab_0    conda-forge
cryptography              35.0.0           py39hfb8cd70_1    conda-forge
cycler                    0.11.0             pyhd8ed1ab_0    conda-forge
debugpy                   1.4.1            py39hfb83b0d_0    conda-forge
decorator                 5.1.0              pyhd8ed1ab_0    conda-forge
defusedxml                0.7.1              pyhd8ed1ab_0    conda-forge
entrypoints               0.3             pyhd8ed1ab_1003    conda-forge
freetype                  2.10.4               h17b34a0_1    conda-forge
future                    0.18.2           py39h2804cbe_3    conda-forge
idna                      3.1                pyhd3deb0d_0    conda-forge
importlib-metadata        4.8.1            py39h2804cbe_0    conda-forge
ipykernel                 6.4.2            py39h32adebf_0    conda-forge
ipython                   7.29.0           py39h32adebf_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
jbig                      2.1               h3422bc3_2003    conda-forge
jedi                      0.18.0           py39h2804cbe_2    conda-forge
jinja2                    3.0.2              pyhd8ed1ab_0    conda-forge
jpeg                      9d                   h27ca646_0    conda-forge
json5                     0.9.5              pyh9f0ad1d_0    conda-forge
jsonschema                4.1.2              pyhd8ed1ab_0    conda-forge
jupyter_client            7.0.6              pyhd8ed1ab_0    conda-forge
jupyter_core              4.9.1            py39h2804cbe_0    conda-forge
jupyter_server            1.11.1             pyhd8ed1ab_0    conda-forge
jupyterlab                3.2.1              pyhd8ed1ab_0    conda-forge
jupyterlab_pygments       0.1.2              pyh9f0ad1d_0    conda-forge
jupyterlab_server         2.8.2              pyhd8ed1ab_0    conda-forge
kiwisolver                1.3.2            py39h4d2d688_0    conda-forge
lcms2                     2.12                 had6a04f_0    conda-forge
lerc                      3.0                  hbdafb3b_0    conda-forge
libblas                   3.9.0           12_osxarm64_openblas    conda-forge
libcblas                  3.9.0           12_osxarm64_openblas    conda-forge
libcxx                    12.0.1               h168391b_0    conda-forge
libdeflate                1.8                  h3422bc3_0    conda-forge
libffi                    3.4.2                hbdafb3b_4    conda-forge
libgfortran               5.0.0.dev0      11_0_1_hf114ba7_23    conda-forge
libgfortran5              11.0.1.dev0         hf114ba7_23    conda-forge
liblapack                 3.9.0           12_osxarm64_openblas    conda-forge
libopenblas               0.3.18          openmp_h5dd58f0_0    conda-forge
libpng                    1.6.37               hf7e6567_2    conda-forge
libprotobuf               3.18.1               hccf11d3_0    conda-forge
libsodium                 1.0.18               h27ca646_1    conda-forge
libtiff                   4.3.0                h74060c4_2    conda-forge
libwebp-base              1.2.1                h3422bc3_0    conda-forge
libzlib                   1.2.11            hee7b306_1013    conda-forge
llvm-openmp               12.0.1               hf3c4609_1    conda-forge
lz4-c                     1.9.3                hbdafb3b_1    conda-forge
markupsafe                2.0.1            py39h5161555_0    conda-forge
matplotlib                3.4.3            py39hdf13c20_1    conda-forge
matplotlib-base           3.4.3            py39h5aa4fe7_1    conda-forge
matplotlib-inline         0.1.3              pyhd8ed1ab_0    conda-forge
mistune                   0.8.4           py39h5161555_1004    conda-forge
nbclassic                 0.3.4              pyhd8ed1ab_0    conda-forge
nbclient                  0.5.4              pyhd8ed1ab_0    conda-forge
nbconvert                 6.2.0            py39h2804cbe_0    conda-forge
nbformat                  5.1.3              pyhd8ed1ab_0    conda-forge
ncurses                   6.2                  h9aa5885_4    conda-forge
nest-asyncio              1.5.1              pyhd8ed1ab_0    conda-forge
ninja                     1.10.2               hc021e02_1    conda-forge
notebook                  6.4.5              pyha770c72_0    conda-forge
numpy                     1.21.3           py39h1f3b974_0    conda-forge
olefile                   0.46               pyh9f0ad1d_1    conda-forge
openjpeg                  2.4.0                h062765e_1    conda-forge
openssl                   1.1.1l               h3422bc3_0    conda-forge
packaging                 21.2               pyhd8ed1ab_1    conda-forge
pandas                    1.3.4            py39h7f752ed_0    conda-forge
pandocfilters             1.5.0              pyhd8ed1ab_0    conda-forge
parso                     0.8.2              pyhd8ed1ab_0    conda-forge
pexpect                   4.8.0              pyh9f0ad1d_2    conda-forge
pickleshare               0.7.5                   py_1003    conda-forge
pillow                    8.3.2            py39ha74c66e_0    conda-forge
pip                       21.3.1             pyhd8ed1ab_0    conda-forge
prometheus_client         0.12.0             pyhd8ed1ab_0    conda-forge
prompt-toolkit            3.0.21             pyha770c72_0    conda-forge
ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
pycparser                 2.20               pyh9f0ad1d_2    conda-forge
pygments                  2.10.0             pyhd8ed1ab_0    conda-forge
pyopenssl                 21.0.0             pyhd8ed1ab_0    conda-forge
pyparsing                 2.4.7              pyhd8ed1ab_1    conda-forge
pyrsistent                0.17.3           py39h5161555_2    conda-forge
pysocks                   1.7.1            py39h2804cbe_3    conda-forge
python                    3.9.7           h54d631c_3_cpython    conda-forge
python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
python_abi                3.9                      2_cp39    conda-forge
pytorch                   1.10.0          cpu_py39h7601aee_0    conda-forge
pytorch-cpu               1.10.0          cpu_py39hd610c6a_0    conda-forge
pytz                      2021.3             pyhd8ed1ab_0    conda-forge
pyzmq                     22.3.0           py39h02c6a76_0    conda-forge
readline                  8.1                  hedafd6a_0    conda-forge
requests                  2.26.0             pyhd8ed1ab_0    conda-forge
requests-unixsocket       0.2.0                      py_0    conda-forge
send2trash                1.8.0              pyhd8ed1ab_0    conda-forge
setuptools                58.3.0           py39h2804cbe_0    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
sleef                     3.5.1                h156473d_2    conda-forge
sniffio                   1.2.0            py39h2804cbe_1    conda-forge
sqlite                    3.36.0               h72a2b83_2    conda-forge
terminado                 0.12.1           py39h2804cbe_0    conda-forge
testpath                  0.5.0              pyhd8ed1ab_0    conda-forge
tk                        8.6.11               he1e0b03_1    conda-forge
torchvision               0.10.1          py39h0a40b5a_0_cpu    conda-forge
tornado                   6.1              py39h5161555_1    conda-forge
traitlets                 5.1.1              pyhd8ed1ab_0    conda-forge
typing_extensions         3.10.0.2           pyha770c72_0    conda-forge
tzdata                    2021e                he74cb21_0    conda-forge
urllib3                   1.26.7             pyhd8ed1ab_0    conda-forge
wcwidth                   0.2.5              pyh9f0ad1d_2    conda-forge
webencodings              0.5.1                      py_1    conda-forge
websocket-client          0.57.0           py39h2804cbe_4    conda-forge
wheel                     0.37.0             pyhd8ed1ab_1    conda-forge
xz                        5.2.5                h642e427_1    conda-forge
zeromq                    4.3.4                hbdafb3b_1    conda-forge
zipp                      3.6.0              pyhd8ed1ab_0    conda-forge
zlib                      1.2.11            hee7b306_1013    conda-forge
zstd                      1.5.0                h861e0a7_0    conda-forge
```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```bash
$ conda info
          conda version : 4.10.3
    conda-build version : not installed
         python version : 3.9.7.final.0
       virtual packages : __osx=12.0.1=0
                          __unix=0=0
                          __archspec=1=arm64

...

               platform : osx-arm64
             user-agent : conda/4.10.3 requests/2.26.0 CPython/3.9.7 Darwin/21.1.0 OSX/12.0.1
                UID:GID : 501:20
             netrc file : None
           offline mode : False

```
</details>

**Comment by ngam** [comment]
@hmaarrfk could you check in your local build (discussed in #31) if this `torchvision.io.read_image` produces any errors? I am surprised the NVIDIA people didn't spot this. I can also install the build in #31 to test it if you would like me to. Thanks!

**Comment by h-vetinari** [comment]
This sounds like a reappearance of https://github.com/pytorch/vision/issues/4181

**Comment by ngam** [comment]
> This sounds like a reappearance of [pytorch/vision#4181](https://github.com/pytorch/vision/issues/4181)

and it was reported in this one: https://github.com/pytorch/vision/issues/4738

**Comment by h-vetinari** [comment]
> The tests for MacOS ARM64 seem to have been successful 👀

FYI, MacOS ARM builds are not being tested in CI (due to lack of native hardware; packages are cross-compiled)

**Comment by ngam** [comment]
> FYI, MacOS ARM builds are not being tested in CI (due to lack of native hardware; packages are cross-compiled)

Ah! Makes sense, thanks for clarifying!

**Comment by h-vetinari** [comment]
Seems that https://github.com/pytorch/vision/commit/16caba5e72d4995a1743c72824b923920060cb26 was not backported to the 0.10 branch. OTOH, that means we should be fine once #31 lands.

**Comment by ngam** [comment]
> Seems that https://github.com/pytorch/vision/commit/16caba5e72d4995a1743c72824b923920060cb26 was not backported to the 0.10 branch. OTOH, that means we should be fine once #31 lands.

ETA?

**Comment by h-vetinari** [comment]
> ETA?

None, but you can help by figuring out the test errors (possibly raising issues upstream).

We could have a new build for 0.10.1 that backports https://github.com/pytorch/vision/commit/16caba5e72d4995a1743c72824b923920060cb26 (assuming it cherry-picks cleanly)

**Comment by ngam** [comment]
Okay, I'll try to help later in the week. I can potentially build from source and play around to see what's going on. Thing is, it's pretty minor bug on the surface, as in one could replace this with a PIL call or something, but I worry about it having downstream effects...

**Comment by ngam** [comment]
> We could have a new build for 0.10.1 that backports https://github.com/pytorch/vision/commit/16caba5e72d4995a1743c72824b923920060cb26 (assuming it cherry-picks cleanly)

Nah, wouldn't go that far...

**Comment by ngam** [comment]
Hi @h-vetinari, I am trying to quickly learn this type of thing so that I can help debug in the future. If you have a moment, could you look at this attempt to build torchaudio and let me know how I can improve? Thanks @hmaarrfk too if you have time :) 

https://github.com/conda-forge/staged-recipes/pull/16720

**Comment by ngam** [comment]
closing this for now, and focusing on 0.11.1 instead. I will attempt to pull request once I have a decent setup working (already built locally). You will need to make sure deps are set up correctly once I get to it

**Comment by ngam** [comment]
Fyi this has been resolved (in 0.11.2). Also lol at my comments from the first time I came across conda-forge stuff.

Thanks, @h-vetinari, for encouraging me to get involved!

> > ETA?
> 
> None, but you can help by figuring out the test errors (possibly raising issues upstream).
> 
> We could have a new build for 0.10.1 that backports [pytorch/vision@16caba5](https://github.com/pytorch/vision/commit/16caba5e72d4995a1743c72824b923920060cb26) (assuming it cherry-picks cleanly)
